independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/core/ServerSerialisers.java

```diff
@@ -502,4 +523,15 @@ public static void encodeResponseHeaders(ResteasyReactiveRequestContext requestC
         }
     }
 
+    private static boolean useSet(String header) {
```

Comment:
This method looks too complex and confusing to me. Maybe rename it to something like: `requireSingleHeader`?

What about an `equalsIgnoreCase` for `Content-Type` and `Content-Length`?

Reply:
Since this will be called for every request, and because `Content-` is common, I wanted to make this as light as possible

Reply:
I renamed the method according to your suggestion